database/migrations/2023_09_07_135157_add_token_number_column_to_nft_activity_table.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+use App\Models\Collection;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('nft_activity', function (Blueprint $table) {
+            $table->addColumn('numeric', 'token_number', ['numeric_type' => 'numeric'])->nullable();
+            $table->foreignIdFor(Collection::class)->nullable()->constrained()->cascadeOnDelete();
+            $table->unsignedBigInteger('nft_id')->nullable()->change();
+
+            // TODO: remove `nft_id`
+
+            $table->unique(['tx_hash', 'collection_id', 'token_number', 'type']);
```

Comment:
Here we also need to include `log_index`. See [this](https://app.clickup.com/t/85ztyhwq4) task.